### PR TITLE
Adding fromversion check in validate

### PR DESCRIPTION
**Pull request by roysagi**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/28764

## Description
Adding check to validate fromversion: 5.0.0 (or above) when running on non-featrue branch or when not comparing to feature branch.

@barchen1 - FYI

### CODE REVIEW

File: TestSuite/assets/default_script/sample_script.yml [hunk 1]
```diff
@@ -17,4 +17,5 @@ args:
   description: sample arg
 scripttarget: 0
 runonce: false
+fromverrsion: 5.0.0
```
Comment (bakatzir) [inline TestSuite/assets/default_script/sample_script.yml:20]:
```suggestion
fromversion: 5.0.0
```

File: demisto_sdk/commands/common/errors.py [hunk 2]
```diff
@@ -238,6 +239,16 @@ def invalid_id_set():
...
 
+    @staticmethod
+    @error_code_decorator
+    def no_minimal_fromversion_in_file(fromversion, oldest_supported_version):
+        if fromversion == 'fromversion':
+            return f"{fromversion} field is invalid.\nAdd `{fromversion}: " \
+                   f"{oldest_supported_version}` to the file."
+        else:
+            return f'{fromversion} field is invalid.\nAdd `"{fromversion}": "{oldest_supported_version}"` ' \
+                   f'to the file.'
```
Comment (bakatzir) [inline demisto_sdk/commands/common/errors.py:250]: what is the difference between the cases?

File: demisto_sdk/commands/common/hook_validations/content_entity_validator.py [hunk 3]
```diff
@@ -186,3 +192,27 @@ def yml_has_test_key(self, test_playbooks: list, file_type: str) -> bool:
...
+                                                                                  OLDEST_SUPPORTED_VERSION)
+                if self.handle_error(error_message, error_code, file_path=self.file_path):
+                    return False
+
+        elif self.file_path.endswith('.yml'):
+            if LooseVersion(self.current_file.get('fromversion', '0.0.0')) < LooseVersion(OLDEST_SUPPORTED_VERSION):
+                error_message, error_code = Errors.no_minimal_fromversion_in_file('fromversion',
+                                                                                  OLDEST_SUPPORTED_VERSION)
+                if self.handle_error(error_message, error_code, file_path=self.file_path):
+                    return False
```
Comment (bakatzir) [inline demisto_sdk/commands/common/hook_validations/content_entity_validator.py:216]:
```suggestion
        if self.file_path.endswith('json'):
            if LooseVersion(self.current_file.get('fromVersion', '0.0.0')) < LooseVersion(OLDEST_SUPPORTED_VERSION):
                error_message, error_code = Errors.no_minimal_fromversion_in_file('fromVersion',
                                                                                  OLDEST_SUPPORTED_VERSION)
                if self.handle_error(error_message, error_code, file_path=self.file_path):
                    return False
        elif self.file_path.endswith('.yml'):
            if LooseVersion(self.current_file.get('fromversion', '0.0.0')) < LooseVersion(OLDEST_SUPPORTED_VERSION):
                error_message, error_code = Errors.no_minimal_fromversion_in_file('fromversion',
                                                                                  OLDEST_SUPPORTED_VERSION)
                if self.handle_error(error_message, error_code, file_path=self.file_path):
                    return False
```

Comment (bakatzir) [reply demisto_sdk/commands/common/hook_validations/content_entity_validator.py:216]: when if/elif, no newline.

File: demisto_sdk/commands/common/errors.py (hunk 2 above)
Comment (roysagi) [reply demisto_sdk/commands/common/errors.py:250]: the use of `"` - I want to be written as a simple copy-paste solution

File: demisto_sdk/commands/common/hook_validations/content_entity_validator.py [hunk 4]
```diff
@@ -186,3 +192,30 @@ def yml_has_test_key(self, test_playbooks: list, file_type: str) -> bool:
...
+
+    def should_run_fromversion_validation(self):
+        # skip check if the comparison is to a feature branch or if you are on the feature branch itself.
+        if any((feature_branch_name in self.prev_ver or feature_branch_name in self.branch_name)
+               for feature_branch_name in FEATURE_BRANCHES):
+            return False
+
+        return True
+
+    def is_valid_fromversion(self):
```
Comment (bakatzir) [inline demisto_sdk/commands/common/hook_validations/content_entity_validator.py:204]: add docstring

**Review by bakatzir — APPROVED** [review]
Nice.

**Comment by roysagi** [comment]
Can only be merged after: https://github.com/demisto/content/pull/9171

**Pull request merged**